packages/common/src/types/auth.ts

```diff
@@ -6,3 +20,41 @@ export enum AuthTokenPrefix {
     SERVICE_ACCOUNT = 'ldsvc_',
     PERSONAL_ACCESS_TOKEN = 'ldpat_',
 }
+
+type EmbeddedAccount = LightdashAuth<{
+    type: 'embed';
+    data: EmbedJwt;
+    source: string;
+}>;
+
+type LightdashUserAccount = LightdashAuth<{
+    type: 'session';
+    source: string;
+}>;
+
+type ServiceAccount = LightdashAuth<{
+    type: 'service-account';
+    source: string;
+}>;
+
+type LightdashAccountAuth =
+    | EmbeddedAccount
+    | LightdashUserAccount
+    | ServiceAccount;
+
+/**
+ * The account object is used to store the account information for the user.
+ * It's meant to be agnostic of the authentication method.
+ */
+export type Account = {
+    organization: Pick<Organization, 'organizationUuid'>;
+    authentication: LightdashAccountAuth;
+    user: SessionUser;
+};
+
+export function getEmbeddedAuth(account: Account): EmbeddedAccount {
```

Comment:
Could also be a typeguard:

```
isEmbeddedAccountAuth(auth: LightdashAccountAuth): auth is EmbeddedAccount {
  ...
};
```

Reply:
this allows you to use this in places where you don't want to throw error and narrow types

Reply:
What if the typeguard function evaluates `false`? Would we not throw?